<?xml version="1.0" encoding="UTF-8"?>
<!--
 QLTOTapeMan - Qt-based LTO Tape Manager

 Copyright (c) 2026 Jeffrey ZHU (zhxsh1225@gmail.com)
 https://github.com/Gypsop/QLTOTapeMan
-->
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>700</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QLTOTapeMan - LTO Tape Manager</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <!-- Device Selection Bar -->
    <item>
     <widget class="QFrame" name="deviceFrame">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <layout class="QHBoxLayout" name="deviceLayout">
       <item>
        <widget class="QLabel" name="lblDevice">
         <property name="text">
          <string>Tape Device:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="comboDevice">
         <property name="minimumWidth">
          <number>300</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnRefresh">
         <property name="text">
          <string>Refresh</string>
         </property>
         <property name="icon">
          <iconset theme="view-refresh"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnConnect">
         <property name="text">
          <string>Connect</string>
         </property>
         <property name="icon">
          <iconset theme="network-connect"/>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDisconnect">
         <property name="text">
          <string>Disconnect</string>
         </property>
         <property name="icon">
          <iconset theme="network-disconnect"/>
         </property>
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="deviceSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>

    <!-- Main Content -->
    <item>
     <widget class="QSplitter" name="mainSplitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>

      <!-- Left Panel: Tape Info -->
      <widget class="QGroupBox" name="groupTapeInfo">
       <property name="title">
        <string>Tape Information</string>
       </property>
       <property name="minimumWidth">
        <number>250</number>
       </property>
       <layout class="QVBoxLayout" name="tapeInfoLayout">
        <item>
         <layout class="QFormLayout" name="tapeFormLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="lblStatusLabel">
            <property name="text">
             <string>Status:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="lblTapeStatus">
            <property name="text">
             <string>Not connected</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblTypeLabel">
            <property name="text">
             <string>Media Type:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLabel" name="lblMediaType">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblCapacityLabel">
            <property name="text">
             <string>Capacity:</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="lblCapacity">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblVolumeLabel">
            <property name="text">
             <string>Volume ID:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QLabel" name="lblVolumeId">
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="tapeInfoSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="tapeButtonLayout">
          <item>
           <widget class="QPushButton" name="btnMount">
            <property name="text">
             <string>Mount</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnEject">
            <property name="text">
             <string>Eject</string>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>

      <!-- Right Panel: File Browser / Transfer -->
      <widget class="QGroupBox" name="groupFiles">
       <property name="title">
        <string>Files</string>
       </property>
       <layout class="QVBoxLayout" name="filesLayout">
        <item>
         <widget class="QTreeView" name="treeFiles">
          <property name="alternatingRowColors">
           <bool>true</bool>
          </property>
          <property name="selectionMode">
           <enum>QAbstractItemView::ExtendedSelection</enum>
          </property>
          <property name="sortingEnabled">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="fileButtonLayout">
          <item>
           <widget class="QPushButton" name="btnWrite">
            <property name="text">
             <string>Write Files to Tape</string>
            </property>
            <property name="icon">
             <iconset theme="go-up"/>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="btnRead">
            <property name="text">
             <string>Read Files from Tape</string>
            </property>
            <property name="icon">
             <iconset theme="go-down"/>
            </property>
            <property name="enabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="fileButtonSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>

  <!-- Menu Bar -->
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="actionOpenIndex"/>
    <addaction name="actionSaveIndex"/>
    <addaction name="actionSaveIndexAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuDevice">
    <property name="title">
     <string>&amp;Device</string>
    </property>
    <addaction name="actionRefreshDevices"/>
    <addaction name="actionDeviceProperties"/>
    <addaction name="separator"/>
    <addaction name="actionFormatTape"/>
    <addaction name="actionEjectTape"/>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="actionDirectReadWrite"/>
    <addaction name="actionVerifyFiles"/>
    <addaction name="separator"/>
    <addaction name="actionSettings"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
    <addaction name="actionAboutQt"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuDevice"/>
   <addaction name="menuTools"/>
   <addaction name="menuHelp"/>
  </widget>

  <!-- Status Bar -->
  <widget class="QStatusBar" name="statusbar"/>

  <!-- Actions -->
  <action name="actionNew">
   <property name="text">
    <string>&amp;New Session</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpenIndex">
   <property name="text">
    <string>&amp;Open Index...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveIndex">
   <property name="text">
    <string>&amp;Save Index</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionSaveIndexAs">
   <property name="text">
    <string>Save Index &amp;As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>E&amp;xit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionRefreshDevices">
   <property name="text">
    <string>&amp;Refresh Devices</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionDeviceProperties">
   <property name="text">
    <string>Device &amp;Properties...</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionFormatTape">
   <property name="text">
    <string>&amp;Format Tape for LTFS...</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionEjectTape">
   <property name="text">
    <string>&amp;Eject Tape</string>
   </property>
   <property name="enabled">
    <bool>false</bool>
   </property>
  </action>
  <action name="actionDirectReadWrite">
   <property name="text">
    <string>&amp;Direct Read/Write...</string>
   </property>
  </action>
  <action name="actionVerifyFiles">
   <property name="text">
    <string>&amp;Verify Files...</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>&amp;Settings...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+,</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="text">
    <string>&amp;About QLTOTapeMan...</string>
   </property>
  </action>
  <action name="actionAboutQt">
   <property name="text">
    <string>About &amp;Qt...</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
